Automattic/wp-calypso
# Help: Cleanup courses of unnecessary state

#### Changes proposed in this Pull Request

This PR cleans up some unnecessary complications around how we were managing help courses. They practically are declared in a constant (and have been for several years already) but were being passed through Redux for some reason. That would have been helpful for fetching them server-side, but if we ever decide to do that, we can just do it with `react-query`. So this PR cleans up all that state and does some additional cleanup. It might be better to review if you look at the changeset by commit. We're essentially doing the following in this PR:

* Remove "fetching" of courses from `Courses` in favor of immediately loading them from constants.
* Move `QueryUserPurchases` up top, ideally query components should be rendered as soon as possible.
* Convert `Courses` to a functional component with hooks.
* Remove unnecessary tests that IMHO were not really testing component logic, but rather the plan and feature definitions.
* Remove all the unused courses state (and its tests).

This PR is also helpful for our work to extract inline help into a standalone widget because it shaves off a needless piece of its Redux state.

This PR also removes an `UNSAFE_` method, see #58453. This is what brought me here originally, actually!

#### Testing instructions

* Go to `/help/courses` with a user that has a business plan.
* Verify courses still load well.
